get-all-systems-with-domains.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+description = """
+Script that provides a listing of all domains found within the system, paired
+with the org and space names they belong to and a bit of other metadata.  The
+output is sorted in chronological order by created_at date.
+
+NOTE:  This script assumes you are logged into CF on the command line as it
+makes use of calls with the cf CLI.
+"""
+
+import argparse
+import json
+import logging
+import subprocess
+
+# CF API URI endpoint configuration.
+CF_SERVICE_INSTANCES_API_URI = "/v3/service_instances?service_plan_names=cdn-route,custom-domain,domain,domain-with-cdn&order_by=created_at"
+CF_SPACES_API_URI = "/v3/spaces/"
+CF_ORGANIZATIONS_API_URI = "/v3/organizations/"
+CF_SERVICE_PLANS_API_URI = "/v3/service_plans/"
+CF_SERVICE_OFFERINGS_API_URI = "/v3/service_offerings/"
+
+
+# Federalist organization name.
+FEDERALIST_ORG_NAME = "gsa-18f-federalist"
+
+
+# Logging configuration - this will output logs to the console/STDOUT/STDERR.
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(levelname)s: %(message)s")
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
+
+
+def main():
+    """
+    Runs the script.
+    """
+
+    args = get_args()
+
+    service_instances = get_all_instances(args.limit)
+    num_records = len(service_instances)
+
+    # Check to see if we should keep cloud.gov and Federalist domains together
+    # or make them separate and output the results.
+    if not args.split:
+        print("\n----- CSV OUTPUT ({0} TOTAL RECORDS) -----\n".format(
```

Comment:
I think we should drop the printed metadata lines, or make them log statements instead. 
The reason is that by printing these, if I run `python get-all-systems-with-domains.py > domains.csv` then `domains.csv` is not a valid csv file.

A user can use `wc` if the want to know how many records there are, or open the csv with excel. If we need to signal to users that output will be in CSV format, we can do that in the help (although I think most folks who run this would probably immediately recognize csv output when they see it)

Reply:
Okay that's definitely fair - be more UNIX-like!  Was getting too fancy with the output. :-)

Reply:
yeah - I should've been clearer about that earlier when I mentioned print :)